packages/editor/core/src/ui/extensions/code/code-block.ts

```diff
@@ -0,0 +1,368 @@
+              // Check if it's an inline code block
+              event.preventDefault();
+              const text = event.clipboardData.getData("text/plain");
+
+              if (!text) {
+                console.error("Pasted text is empty.");
+                return false;
+              }
+
+              const { tr } = view.state;
+              const { $from, $to } = tr.selection;
+
+              if ($from.pos > $to.pos) {
+                console.error("Invalid selection range.");
+                return false;
+              }
+
+              const docSize = tr.doc.content.size;
+              if ($from.pos < 0 || $to.pos > docSize) {
+                console.error("Selection range is out of document bounds.");
+                return false;
+              }
+
+              // Extend the current selection to replace it with the pasted text
+              // wrapped in an inline code mark
+              const codeMark = view.state.schema.marks.code.create();
+              tr.replaceWith($from.pos, $to.pos, view.state.schema.text(text, [codeMark]));
+              view.dispatch(tr);
+              return true;
+            }
+
+            event.preventDefault();
+            const text = event.clipboardData.getData("text/plain");
+            const vscode = event.clipboardData.getData("vscode-editor-data");
+            const vscodeData = vscode ? JSON.parse(vscode) : undefined;
+            const language = vscodeData?.mode;
+
+            if (vscodeData && language) {
+              const { tr } = view.state;
+              const { $from } = tr.selection;
+
+              // Check if the current line is empty
+              const isCurrentLineEmpty = !$from.parent.textContent.trim();
+
+              let insertPos;
+
+              if (isCurrentLineEmpty) {
+                // If the current line is empty, use the current position
+                insertPos = $from.pos - 1;
+              } else {
+                // If the current line is not empty, insert below the current block node
+                insertPos = $from.end($from.depth) + 1;
+              }
+
+              // Ensure insertPos is within document bounds
+              if (insertPos < 0 || insertPos > tr.doc.content.size) {
+                console.error("Invalid insert position.");
+                return false;
+              }
+
+              // Create a new code block node with the pasted content
+              const textNode = view.state.schema.text(text.replace(/\r\n?/g, "\n"));
+              const codeBlock = this.type.create({ language }, textNode);
+              console.log(insertPos, tr.doc.content.size);
+              if (insertPos <= tr.doc.content.size) {
+                tr.insert(insertPos, codeBlock);
+                view.dispatch(tr);
+                return true;
+              }
+
+              return false;
+            } else {
+              // // complicated paste logic, to be handled later
```

Comment:
Please check and remove commented code if not required

Reply:
I'll soon get back to this part and build over the commented out logic, it was getting a bit complex to handle as i'll have to hook into the existing behaviours that comes with tiptap's default paste handler and the Markdown plugin's paste handler.

Could this stay here for now, so that I could build over it later or should I remove it? @rahulramesha

Reply:
Removed it for now 🙌